examples_gbs/run_tutorial_max_clique.py

```diff
@@ -0,0 +1,173 @@
+
+TA_subgraphs = gbs.sample.to_subgraphs(TA_graph, TA)
+
+counts = TA.counts()
+samples = [TA_subgraphs[i] for i in range(len(TA)) if counts[i] == 8]
+print(len(samples))
+
+##############################################################################
+# GBS produces samples that correspond to subgraphs of large density. For fun, let's confirm this
+# by comparing the average subgraph density in the GBS samples to uniformly generated samples:
+
+GBS_dens = []
+u_dens = []
+
+for s in samples:
+    uniform = list(np.random.choice(24, 8))  # generates uniform samples
+    GBS_subgraph = TA_graph.subgraph(s)
+    uni_subgraph = TA_graph.subgraph(uniform)
+    GBS_dens.append(nx.density(GBS_subgraph))
+    u_dens.append(nx.density(uni_subgraph))
+
+print('GBS mean density = {:.4f}'.format(np.mean(GBS_dens)))
+print('Uniform mean density = {:.4f}'.format(np.mean(u_dens)))
+
+##############################################################################
+# Those look like great GBS samples 💪! To obtain cliques, we shrink the samples by greedily
+# removing nodes with low degree until a clique is found. We can also verify that the first sample
+# got shrunk to a clique:
+
+shrunk = [gbs.clique.shrink(s, TA_graph) for s in samples]
+print(gbs.clique.is_clique(TA_graph.subgraph(shrunk[0])))
+
+##############################################################################
+# Let's take a look at some of these cliques. What are the clique sizes in the first ten samples?
+# What is the average clique size? How about the largest and smallest clique size?
+
+clique_sizes = [len(s) for s in shrunk]
+print('First ten clique sizes = ', clique_sizes[:10])
+print('Average clique size = {:.3f}'.format(np.mean(clique_sizes)))
+print('Maximum clique size = ', np.max(clique_sizes))
+print('Minimum clique size = ', np.min(clique_sizes))
+
+##############################################################################
+# Even in the first few samples, we've already identified larger cliques than the maximal clique
+# we studied before. Indeed, this simple strategy of shrinking samples gives cliques with average
+# size of roughly five.
+#
+# We can improve this by searching in the neighbourhood of these cliques for larger cliques.
+# We'll do this by taking ten iterations of local search and studying the results. Note: this may
+# take a few seconds.
+
+searched = [gbs.clique.search(s, TA_graph, 10) for s in shrunk]
+clique_sizes = [len(s) for s in searched]
+print('First two cliques = ', searched[:2])
+print('Average clique size = {:.3f}'.format(np.mean(clique_sizes)))
+
+##############################################################################
+# Wow! Local search is very helpful, allowing us to quickly identify larger cliques in
+# the vicinity of the smaller cliques. Looking at the cliques we've found, we learn that the largest
+# cliques in the graph have size eight, and there are many of them. Let's take a look at the first

```

Comment:
We can add a line to print the largest clique we have found:
print('Largest clique = ', len(max(searched)))

Reply:
If you look at the average clique size, which I think is 7.999, it shows that basically all of them have size eight, so I think it's not so useful to find that the largest one has size eight